modules/standard/SysError.chpl

```diff
@@ -41,17 +41,342 @@ module SysError {
 
 use SysBasic;
 
+/*
+
+   ``SystemError`` is a base class for ``Errors`` generated from ``syserr``.
+   It provides factory methods to create different subtypes based on the
+   ``syserr`` that is passed.
+
+*/
 class SystemError : Error {
-  var err: syserr;
+  var err:     syserr;
+  var details: string;
+
+  proc SystemError(err: syserr, details: string = "") {
+    this.err     = err;
+    this.details = details;
```

Comment:
FYI. I think this kind of spacing-lining-up can quickly become more trouble than it's worth. I usually try to reserve that effort for limited cases where the spacing makes comparing different lines easier. That said, I don't mind if you do it.

Reply:
Yeah this was probably overkill.